### ISSUE TITLE
is the onOutputreceived callback supposed to be called for a viewonly wallet?

### ISSUE THREAD

**Issue by spirobel** [monero-ecosystem/monero-javascript#105]
is the onOutputreceived callback supposed to be called for a viewonly wallet?
I am only seeing calls to onBalancesChanged, but the onOutputReceived callback does not get called when I receive money. 
Is this expected behavior, or should the onOutputReceived callback also be called?
OnNewBlock also works fine.

**Comment by woodser** [comment]
Yes I'm seeing `onOutputReceived()` is called correctly for view-only wallets (when receiving funds from a different source account).

**Comment by spirobel** [comment]
hm... for me it does not call `onOutputReceived()` while the other callbacks work fine. Do you look at the wasm wallet or the rpc wallet? I also copy and pasted the code directly from a known working webapp that I wrote earlier. 
I also see the transactions show up when I do a call to `getTxs()`. I spent a while trying all kinds of random things to get it to work, but maybe a different approach is better. I could also use the `onNewBlock()` callback to do a `getTx()` for the last 10 blocks and supply the clients with this information instead. (I am writing a small tool similar to the wallet-rpc using expressjs and the monero-javascript library)

**Comment by spirobel** [comment]
I use the wasm wallet with the latest 0.7.4 version of monero-javascript. I will go the getTxs() route for now, it has the added advantage that the clients receive updates on the numbers of confirmations as they happen and it seems more robust.

**Comment by woodser** [comment]
It's very odd it doesn't work. I tested specifically that it's triggered at 0 confirmations and 1 confirmation. If you're able to isolate the issue or post something reproducible, I'm happy to take a look.
